examples/new_theme.py

```diff
@@ -0,0 +1,36 @@
+"""
+Displays an image and sets the theme to 'light'.
+"""
+
+from skimage import data
+import napari
+from napari.utils.theme import Palette, available_themes
+
+with napari.gui_qt():
+    # create the viewer with an image
+    viewer = napari.view_image(
+        data.astronaut(), rgb=True, name='astronaut'
+    )
+
+    # Create a new palette. Note we must use either the `light` or
+    # `dark` folder to avoid needing to build new icons
+    blue_palette = Palette(
+        folder='dark',
+        background='rgb(28, 31, 48)',
+        foreground='rgb(45, 52, 71)',
+        primary='rgb(80, 88, 108)',
+        secondary='rgb(134, 142, 147)',
+        highlight='rgb(106, 115, 128)',
+        text='rgb(240, 241, 242)',
+        icon='rgb(209, 210, 212)',
+        warning='rgb(153, 18, 31)',
+        current='rgb(184, 112, 0)',
+        syntax_style='native',
+        console='rgb(0, 0, 0)',
+        canvas='black',
+    )
```

Comment:
@sofroniewn Bringing some of our Zulip discussion here. I also chatted a bit with tlambert03 . We both agree that dicts (Talley suggests TypedDict but I'll let him elaborate on that) are preferable to a NamedTuple. Regarding dicts being mutable, our suggestion is that users should never have to interact with the actual dicts — they only get existing themes through a convenience function that, importantly, returns a copy (documented as a copy and read-only, so users don't think that they can simply change keys in there). Users only interact with themes as:

- viewer.theme = name
- napari.utils.available_themes
- napari.utils.register_theme(name, dict)
- napari.utils.get_theme(name)

Dicts are simple objects that everyone understands. Additionally, dicts are easy to combine, see:

https://treyhunner.com/2016/02/how-to-merge-dictionaries-in-python/

And new in Python 3.9:

https://www.python.org/dev/peps/pep-0584/

So it would be easy to do:

my_theme = {**get_theme('dark'),  **my_updates}

Or, in 3.9+:

my_theme = get_theme('dark') | my_updates

That allows partial building and combination of themes, which seems trickier with NamedTuples. 

Meanwhile, I think the above public API suitably addresses concerns about mutability.